File: common/src/test/java/com/fasterxml/jackson/module/scala/deser/JavaInteropTest.java
package com.fasterxml.jackson.module.scala.deser;

import com.fasterxml.jackson.core.JsonParseException;
import org.junit.Test;

import java.io.IOException;

public class JavaInteropTest {

    @Test
    public void testInteropDeserialization() throws IOException {
        B v = Util.mapper().readValue(Util.jsonString(), B.class);
        assert(v.equals(new B("asdf", new A1("qwer"))));
    }

}

File: src/test/scala/tools/jackson/module/scala/deser/JavaInteropTest.scala
package tools.jackson.module.scala.deser

import tools.jackson.module.scala.BaseSpec

class JavaInteropTest extends BaseSpec {
  "Scala module" should "interoperate with Java classes" in {
    val v = Util.mapper.readValue(Util.jsonString, classOf[B])
    v shouldEqual new B("asdf", new A1("qwer"))
  }
}
